core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1298,7 +1329,7 @@ private[spark] class DAGScheduler(
     // `findMissingPartitions()` returns all partitions every time.
     stage match {
       case sms: ShuffleMapStage if stage.isIndeterminate && !sms.isAvailable =>
-        mapOutputTracker.unregisterAllMapOutput(sms.shuffleDep.shuffleId)
+        mapOutputTracker.unregisterAllMapAndMergeOutput(sms.shuffleDep.shuffleId)
```

Comment:
If we make this change, is SPARK-32923 (for properly handling indeterminate stage retries) still needed as part of SPARK-30602?
This will always recompute all partitions.
Should we also reset the other metadata here, such as resetting `sms.shuffleDep.shuffleMergeEnabled`?
This way it would make sure that the later invocation to `prepareShuffleServicesForShuffleMapStage` would not be interfered from the previous attempt of this indeterminate stage.

Reply:
Made a comment here to handle clean up of shuffle merge metadata as part of handling SPARK-32923 (non-deterministic stage retries) and in SPARK-35547 (handling barrier execution mode)